### PR TITLE
gh-146310: Fix ensurepip to treat empty WHEEL_PKG_DIR as unset

### LINKED ISSUE #146310: ensurepip looks for wheels in CWD when WHEEL_PKG_DIR is an empty string
### Bug description:

Until and in 3.12, `ensurepip` stored the `sysconfig.get_config_var('WHEEL_PKG_DIR')` result verbatim as a string in `_WHEEL_PKG_DIR`, and used `os` functionality to list wheels in the specified directory.

Since 3.13 (tested on 3.14 but I also reviewed the 3.13 library code), the config value (when not `None`) is normalized to a `pathlib.Path`, and the `.glob` method of that is used instead.

This causes a different behaviour when the returned value (as is the case for my local source builds) is an empty string (`''`, not None):

```python
>>> import os
>>> os.listdir('')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
FileNotFoundError: [Errno 2] No such file or directory: ''

>>> from pathlib import Path
>>> Path('')
PosixPath('.')
>>> Path('').glob('*')
<generator object Path.glob at 0x7c06678cc590>
```

Therefore, `ensurepip` will now bootstrap from the process' CWD, if it contains a suitable pip wheel. And a renamed file that is *not* a valid pip wheel can theoretically just crash:

```
$ touch pip-is-not-in-here.whl
$ py3.14 -m ensurepip
Traceback (most recent call last):
  File "<string>", line 6, in <module>
    runpy.run_module("pip", run_name="__main__", alter_sys=True)
    ~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "<frozen runpy>", line 222, in run_module
  File "<frozen runpy>", line 142, in _get_module_details
ImportError: No module named pip
Traceback (most recent call last):
  File "<frozen runpy>", line 198, in _run_module_as_main
  File "<frozen runpy>", line 88, in _run_code
  File "/opt/python/standard/lib/python3.14/ensurepip/__main__.py", line 5, in <module>
    sys.exit(ensurepip._main())
             ~~~~~~~~~~~~~~~^^
  File "/opt/python/standard/lib/python3.14/ensurepip/__init__.py", line 257, in _main
    return _bootstrap(
        root=args.root,
    ...<4 lines>...
        default_pip=args.default_pip,
    )
  File "/opt/python/standard/lib/python3.14/ensurepip/__init__.py", line 172, in _bootstrap
    return _run_pip([*args, "pip"], [os.fsdecode(tmp_wheel_path)])
  File "/opt/python/standard/lib/python3.14/ensurepip/__init__.py", line 87, in _run_pip
    return subprocess.run(cmd, check=True).returncode
           ~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "/opt/python/standard/lib/python3.14/subprocess.py", line 577, in run
    raise CalledProcessError(retcode, process.args,
                             output=stdout, stderr=stderr)
subprocess.CalledProcessError: Command '['/usr/local/bin/py3.14', '-W', 'ignore::DeprecationWarning', '-c', '\nimport runpy\nimport sys\nsys.path = [\'/tmp/tmpc9p5g7z_/pip-is-not-in-here.whl\'] + sys.path\nsys.argv[1:] = [\'install\', \'--no-cache-dir\', \'--no-index\', \'--find-links\', \'/tmp/tmpc9p5g7z_\', \'pip\']\nrunpy.run_module("pip", run_name="__main__", alter_sys=True)\n']' returned non-zero exit status 1.
```

Actually this behaviour is useful for me (I want to test bootstrapping some modified versions of pip), but it does seem to represent an un-documented change that could break things for someone else.

(Actually I wonder if it is really `os.listdir('')` that should have been changed forever ago. After all, a `''` entry in `sys.path` does work!)

### CPython versions tested on:

3.14

### Operating systems tested on:

_No response_

### Linked PRs
* gh-146357
* gh-146534
* gh-146535

### PR DESCRIPTION
**Pull request by kimimgo**
Fixes #146310

`Path('')` resolves to the current working directory (`PosixPath('.')`), so when `WHEEL_PKG_DIR` is an empty string (not `None`), `ensurepip` searches CWD for wheel files. This can cause unexpected behavior or crashes if CWD contains files matching `pip-*.whl`.

## Changes

- `Lib/ensurepip/__init__.py`: Add truthiness check so empty strings are treated the same as `None`
- `Lib/test/test_ensurepip.py`: Add test verifying empty `WHEEL_PKG_DIR` does not search CWD

## Before/After

```python
# Before: Path('').resolve() -> PosixPath('/current/working/dir')
# Empty string was treated as a valid directory path

# After: Empty string treated as None -> use bundled wheel
```

* Issue: gh-146310

### CODE REVIEW

File: Lib/ensurepip/__init__.py [hunk 1]
```diff
@@ -16,7 +16,7 @@
 # policies recommend against bundling dependencies. For example, Fedora
 # installs wheel packages in the /usr/share/python-wheels/ directory and don't
 # install the ensurepip._bundled package.
-if (_pkg_dir := sysconfig.get_config_var('WHEEL_PKG_DIR')) is not None:
+if (_pkg_dir := sysconfig.get_config_var('WHEEL_PKG_DIR')) is not None and _pkg_dir:
```
Comment (vstinner) [inline Lib/ensurepip/__init__.py]: You may write it as:

```suggestion
_pkg_dir = sysconfig.get_config_var('WHEEL_PKG_DIR')
if _pkg_dir:
```

File: Lib/test/test_ensurepip.py [hunk 2]
```diff
@@ -36,6 +36,13 @@ def test_version_no_dir(self):
             # when the bundled pip wheel is used, we get _PIP_VERSION
             self.assertEqual(ensurepip._PIP_VERSION, ensurepip.version())
 
+    def test_empty_wheel_pkg_dir_treated_as_none(self):
+        # GH#146310: empty string WHEEL_PKG_DIR should not search CWD.
+        # An empty WHEEL_PKG_DIR converts to Path('.') which would
+        # incorrectly search the current working directory.
+        with unittest.mock.patch.object(ensurepip, '_WHEEL_PKG_DIR', None):
+            self.assertIsNone(ensurepip._find_wheel_pkg_dir_pip())
```
Comment (vstinner) [inline Lib/test/test_ensurepip.py]: This test is not directly related to the fix. I suggest a different test:

```py
    def test_wheel_pkg_dir_none(self):
        for value in ('', None):
            with unittest.mock.patch('sysconfig.get_config_var',
                                     return_value=value) as get_config_var:
                module = import_helper.import_fresh_module('ensurepip')
                self.assertIsNone(module._WHEEL_PKG_DIR)
                get_config_var.assert_called_once_with('WHEEL_PKG_DIR')
```

with `from test.support import import_helper` at top level.

File: Misc/NEWS.d/next/Library/2026-03-24-03-49-50.gh-issue-146310.WhlDir.rst [hunk 3]
```diff
@@ -0,0 +1,2 @@
+Fix :mod:`ensurepip` to treat an empty string ``WHEEL_PKG_DIR`` as unset,
+preventing it from searching the current working directory for wheel files.
```
Comment (vstinner) [inline Misc/NEWS.d/next/Library/2026-03-24-03-49-50.gh-issue-146310.WhlDir.rst]: This NEWS entry sounds too technical. I suggest to describe instead the old behavior:

```
The :mod:`ensurepip` module no longer looks for `pip-*.whl` wheel packages in the current directory.
```

File: Lib/test/test_ensurepip.py [hunk 4]
```diff
@@ -36,6 +37,15 @@ def test_version_no_dir(self):
             # when the bundled pip wheel is used, we get _PIP_VERSION
             self.assertEqual(ensurepip._PIP_VERSION, ensurepip.version())
 
+    def test_wheel_pkg_dir_none(self):
+        # GH#146310: empty or None WHEEL_PKG_DIR should not search CWD
```
Comment (vstinner) [inline Lib/test/test_ensurepip.py]: Usually in Python tests, we use `gh-xxx` format:

```suggestion
        # gh-146310: empty or None WHEEL_PKG_DIR should not search CWD
```

**Review by vstinner — APPROVED** [review]
LGTM
